compiler/importer.nim

```diff
@@ -68,6 +68,22 @@ proc rawImportSymbol(c: PContext, s: PSym) =
     if s.kind == skConverter: addConverter(c, s)
     if hasPattern(s): addPattern(c, s)
 
+proc maybeImportForType(c: PContext, s: PSym, t: PType) =
+  ## Imports a symbol only if it has a strict relation with a type:
+  ##
+  ##  - Proc refers to the type on any of its input or output values.
+  ##
+  ## TODO: generics are not matched currently.
```

Comment:
Yeah, this should be done though. Requires a bit more code, will show you how to do that later.